File: problem/Shortest_Distance_Target/closestTarget.c
int closestTarget(char **words, int wordsSize, char *target, int startIndex)
{
    int n = wordsSize;

    for (int i = 0; i <= n / 2; i++)
    {
        if (strcmp(words[(startIndex + i) % n], target) == 0 ||
            strcmp(words[(startIndex - i + n) % n], target) == 0)
        {
            return i;
        }
    }

    return -1;
}

File: problem/Shortest_Distance_Target/closestTarget.js
/**
 * @param {string[]} words
 * @param {string} target
 * @param {number} startIndex
 * @return {number}
 */
const closestTarget = (words, target, s) => {
    const n = words.length;
    for (let i = 0; i <= n >> 1; i++)
        if (words.at((s + i) % n) === target |
            words.at((s - i) % n) === target)
            return i;

    return -1;
};